### ISSUE TITLE
CustomAttributeTypedArgument.ToString() is missing an open curly

### ISSUE THREAD

**Issue by romkatv** [dotnet/runtime#61282]
### Description

`CustomAttributeTypedArgument.ToString()` produces malformed output if the argument is an array.

### Reproduction Steps

```csharp
Console.WriteLine(
    new System.Reflection.CustomAttributeTypedArgument(
        new [] { new System.Reflection.CustomAttributeTypedArgument(0) }));
```

### Expected behavior

```text
new CustomAttributeTypedArgument[1] { 0 }
```

### Actual behavior

```text
new CustomAttributeTypedArgument[1]0 }
```

### Regression?

It's a regression. The bug was introduced after the .NET 5 release: https://github.com/dotnet/runtime/commit/0c793920f62dd2711e6423a3425110e5cb7acfde#diff-f43b9f98fddac0e7fa971362f6f07f5fc9a3a0a41b485e9df15df87c99d638cb

### Configuration

Works before https://github.com/dotnet/runtime/commit/0c793920f62dd2711e6423a3425110e5cb7acfde. Doesn't work after.

**Comment by ShreyasJejurkar** [comment]
Not able to reproduce it on .NET 5.0.6.

Check below. 
https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK4wD4AEBMAGAWAChcBGY4rUgTgAoBRGANymQHsYwBTGBAOgBqXZAGcoHAJQBuCiRq1iAAmWKYXAO6KqfAEpcAZgBsuAYwTiYfAMKoRCNmACCCBMijBUCLgBUAngAcuABNHZABzVG5eBSIVONUNRQBtAF1FAG8EzW09I1NzDmtbeycXNw8vP0CQ8MieBFo8CUUAXwlpYiA

**Comment by buyaa-n** [comment]
> It's a regression. The bug was introduced after the .NET 5 release

Thanks for reporting @romkatv seems it is introduced in 6.0 preview 4

**Comment by romkatv** [comment]
> seems it is introduced in 6.0 preview 4

The bug was introduced in https://github.com/dotnet/runtime/commit/0c793920f62dd2711e6423a3425110e5cb7acfde.

This should fix it:

```diff
diff --git a/src/libraries/System.Private.CoreLib/src/System/Reflection/CustomAttributeTypedArgument.cs b/src/libraries/System.Private.CoreLib/src/System/Reflection/CustomAttributeTypedArgument.cs
index 18d033b5..d6498fa1 100644
--- a/src/libraries/System.Private.CoreLib/src/System/Reflection/CustomAttributeTypedArgument.cs
+++ b/src/libraries/System.Private.CoreLib/src/System/Reflection/CustomAttributeTypedArgument.cs
@@ -68,6 +68,8 @@ internal string ToString(bool typed)
                 result.Append(count.ToString());
                 result.Append(']');
 
+                result.Append(" { ");
+
                 for (int i = 0; i < count; i++)
                 {
                     if (i != 0)
```

**Comment by buyaa-n** [comment]
Right that can fix, or just change that char `result.Append(']');` into string `result.Append("] { ");`, feel free to fix with corresponding unit test

**Comment by romkatv** [comment]
> feel free to fix with corresponding unit test

I'll leave it up to the professionals.

**Comment by buyaa-n** [comment]
Reopening as it needs to be ported into 6.0 CC @jeffhandley

**Comment by buyaa-n** [comment]
Fix merged into 6.0
